### PR TITLE
feat(taskmanager) spawn tokio tasks with names

### LINKED ISSUE #3985: Use named tokio tasks for spawned critical tasks
### Describe the feature

tokio supports naming tasks which is currently unstable and requires tokio_unstable feature

ref https://github.com/paradigmxyz/reth/blob/b05b84823a9d0f8f3348da346bd95f5456ac6fc3/crates/tasks/src/lib.rs#L321-L324

wdyt @onbjerg ?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xprames**
should close #3985 if we decide to take on the `tokio-unstable` dep

### CODE REVIEW

**Comment by 0xprames** [comment]
```
GitHub
Actions
/ code lint
failed to resolve: could not find `Builder` in `task`
```

linter fails because im not sure it's loading the necessary `RUSTFLAGS` to support the `tokio::task::Builder` in the `tracing` feature crate

I added a `.cargo/config.toml` to this PR with the necessary rustc flags following: https://docs.rs/tokio/latest/tokio/index.html#unstable-features

but not sure how to get the PR linter to pick this up

**Review by mattsse — CHANGES_REQUESTED** [review]
afaik this has now perf implications 

wdyt @onbjerg

File: crates/tasks/src/lib.rs [hunk 1]
```diff
@@ -251,15 +251,24 @@ impl TaskExecutor {
     }
 
     /// Spawns a future on the tokio runtime depending on the [TaskKind]
-    fn spawn_on_rt<F>(&self, fut: F, task_kind: TaskKind) -> JoinHandle<()>
+    fn spawn_on_rt<F>(&self, fut: F, task_kind: TaskKind, name: &'static str) -> JoinHandle<()>
```
Comment (onbjerg) [inline crates/tasks/src/lib.rs:254]: Please make this an `Option` instead to make the behavior of no name more explicit

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Comment by onbjerg** [comment]
@mattsse it makes sense to name the threads so I think this is ok - however, I'd like for you to expand a bit on the perf implications, what are they?

**Comment by onbjerg** [comment]
@0xprames it seems that `tokio::task::Builder` requires both `tokio_unstable` and `tracing` which is probably why lint fails? ref https://docs.rs/tokio/latest/tokio/task/struct.Builder.html

**Comment by 0xprames** [comment]
ah ok if we turn on tracing via the feature crate/`tokio-unstable` flag then every task we spawn (even as is without the name, i.e for non-critical tasks) we will generate a trace for which will probably impact performance.

There's no way to only create traces for critical named tasks looks like

https://github.com/tokio-rs/tokio/blob/efe3ab679a05f3da3fcc511a44120239830254f2/tokio/src/runtime/blocking/pool.rs#L364

https://github.com/tokio-rs/tokio/blob/efe3ab679a05f3da3fcc511a44120239830254f2/tokio/src/runtime/handle.rs#L328

the futures are wrapped or instructed with a `Trace`

**Comment by 0xprames** [comment]
following up re linter: this is the same `cargo clippy` command run locally
https://pastebin.com/S2AK53hk

few warnings, but completes without the compiler error seen in the linter. I think if we add this feature and unstable dep we need to modify the linter config - was what I was alluding to with my original comment above

**Comment by rkrasiuk** [comment]
closing in favor of #4003

**Pull request closed without merging**